deltacat/compute/compactor/steps/repartition.py

```diff
@@ -117,12 +118,18 @@ def repartition_range(
                     & (pc.field(col_name_int64) <= pc.scalar(upper_limit))
                 )
             )
+            if i == 0:
+                partitioned_tables_list[i].append(null_row_table)
+
     partition_table_length = 0
     # After re-grouping the tables by specified ranges, for each group, we need concat and stage the tables
     partition_deltas: List[Delta] = []
     for partition_tables in partitioned_tables_list:
         if len(partition_tables) > 0:
-            partition_table: pa.Table = pa.concat_tables(partition_tables)
+            partition_table: pa.Table = pa.concat_tables(
+                partition_tables
+            ).remove_column(0)
```

Comment:
should we use `drop` method with actual column name created above? Just to make sure we don't drop any actual column from table?

Reply:
good idea

Reply:
turned out drop_columns/drop is not supported in our current version 10.0.1; I'll just use remove_column but by first locating the index with the column name;

Reply:
Then how does this line work? https://github.com/ray-project/deltacat/blob/037b584d7b7330dca6be94adaf279ff01593fee8/deltacat/compute/compactor/steps/hash_bucket.py#L59

Reply:
updated.